dotty-tensorflow/src/main/scala/me.shadaj.scalapy.tensorflow/example/GradientDescentOptimizerExample.scala

```diff
@@ -31,17 +30,9 @@ object GradientDescentOptimizerExample extends Runnable {
     val opt = tf.keras.optimizers.SGD(learning_rate = 0.1, momentum = 0.9)
 
     // Function to calculate gradients
-    def grad(): (Tensor, Seq[Tensor]) = {
-      val tape = tf.GradientTape()
-      // init tape
-      tape.__enter__()
-
-      tape.watch(W)
-      tape.watch(b)
+    def grad(): (Tensor, Seq[Tensor]) = ContextManager.withContext(tf.GradientTape()){ tape =>
```

Comment:
Nit/unrel: scalafmt would be useful in this project (but that's for a separate PR)